src/os/main.js

```diff
@@ -56,30 +57,37 @@ os.Module.configureModule = function($routeProvider) {
 
 os.Module.config(os.Module.configureModule);
 
-
 /**
  * Load the settings, then manually bootstrap angular.
  * @todo should we display an informative error message if there are no settings?
  */
 (function() {
-  if (os.isMainWindow()) {
-    try {
-      // the main application doesn't care what opened it, and accessing window.opener may cause an exception. get rid
-      // of it so things like os.instanceOf don't try using it.
-      window.opener = null;
-    } catch (e) {
-      // this doesn't seem to fail in any browser, but in the off chance it does don't break everything
-    }
+  var appWait = new os.AppStartDelay(
+    function() {
+      return !!osasm.geodesicInverse;
+    },
+    function() {
+      if (os.isMainWindow()) {
+        try {
+          // the main application doesn't care what opened it, and accessing window.opener may cause an exception. get rid
+          // of it so things like os.instanceOf don't try using it.
+          window.opener = null;
+        } catch (e) {
+          // this doesn't seem to fail in any browser, but in the off chance it does don't break everything
+        }
+
+        os.logWindow = new os.debug.FancierWindow('os');
 
-    os.logWindow = new os.debug.FancierWindow('os');
+        // set up request handlers
+        os.net.addDefaultHandlers();
 
-    // set up request handlers
-    os.net.addDefaultHandlers();
+        // initialize settings for this app
+        var settingsInitializer = new os.config.SettingsInitializer();
+        settingsInitializer.init();
+      }
+    });
```

Comment:
This should have an error callback that throws an error, so if anyone encounters the issue they have something useful to tell support (assuming they checked the console).